scripts/depth/trainer.py

```diff
@@ -0,0 +1,499 @@
+                for frame_id in self.opt.frame_ids[1:]:
+                    pred = inputs[("color", frame_id, source_scale)]
+                    identity_reprojection_losses.append(
+                        self.compute_reprojection_loss(pred, target))
+
+                identity_reprojection_losses = mx.nd.concat(*identity_reprojection_losses, dim=1)
+
+                if self.opt.avg_reprojection:
+                    identity_reprojection_loss = \
+                        identity_reprojection_losses.mean(axis=1, keepdims=True)
+                else:
+                    # save both images, and do min all at once below
+                    identity_reprojection_loss = identity_reprojection_losses
+
+            elif self.opt.predictive_mask:
+                # TODO: use predictive_mask
+                pass
+
+            if self.opt.avg_reprojection:
+                reprojection_loss = reprojection_losses.mean(axis=1, keepdims=True)
+            else:
+                reprojection_loss = reprojection_losses
+
+            if not self.opt.disable_automasking:
+                # add random numbers to break ties
+                identity_reprojection_loss = \
+                    identity_reprojection_loss + \
+                    mx.nd.random.randn(*identity_reprojection_loss.shape).as_in_context(
+                        identity_reprojection_loss.context) * 0.00001
+
+                combined = mx.nd.concat(identity_reprojection_loss, reprojection_loss, dim=1)
+            else:
+                combined = reprojection_loss
+
+            if combined.shape[1] == 1:
+                to_optimise = combined
+            else:
+                to_optimise = mx.nd.min(data=combined, axis=1)
+                idxs = mx.nd.argmin(data=combined, axis=1)
+
+            if not self.opt.disable_automasking:
+                outputs["identity_selection/{}".format(scale)] = (
+                        idxs > identity_reprojection_loss.shape[1] - 1).astype('float')
+
+            loss += to_optimise.mean()
+
+            mean_disp = disp.mean(axis=2, keepdims=True).mean(axis=3, keepdims=True)
+            norm_disp = disp / (mean_disp + 1e-7)
+
+            smooth_loss = get_smooth_loss(norm_disp, color)
+
+            loss = loss + self.opt.disparity_smoothness * smooth_loss / (2 ** scale)
+            total_loss = total_loss + loss
+            losses["loss/{}".format(scale)] = loss
+
+        total_loss = total_loss / self.num_scales
+        losses["loss"] = total_loss
+        return losses
+
+    def compute_metrics(self, inputs, outputs, depth_metrics):
+        """Compute depth metrics, to allow monitoring during training
+
+        This isn't particularly accurate as it averages over the entire batch,
+        so is only used to give an indication of validation performance
+        """
+        depth_pred = outputs[("depth", 0, 0)]
+        depth_pred = mx.nd.clip(
+            mx.nd.contrib.BilinearResize2D(depth_pred, height=375, width=1242),
```

Comment:
Pass the `375` and `1242` as arguments. It is better not to hard code these values in trainer.